File: Pro_ODP_NET_for_Oracle_Database_11g-4541/vb_dot_net/Chapter5VBNET/NestedTables/ProductNestedTableFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Oracle.DataAccess.Client
Imports Oracle.DataAccess.Types

<OracleCustomTypeMappingAttribute("EDZEHOO.PRODUCTNESTEDTABLE")> _
Public Class ProductNestedTableFactory
    Implements IOracleArrayTypeFactory

    Public Function CreateArray(ByVal numElems As Integer) As Array Implements Oracle.DataAccess.Types.IOracleArrayTypeFactory.CreateArray
        Return New [String](numElems) {}
    End Function 'CreateArray

    Public Function CreateStatusArray(ByVal numElems As Integer) As Array Implements Oracle.DataAccess.Types.IOracleArrayTypeFactory.CreateStatusArray
        Return New OracleUdtStatus(numElems) {}
    End Function 'CreateStatusArray

End Class 'ProductNestedTableFactory

File: Pro_ODP_NET_for_Oracle_Database_11g-4541/c_sharp/Chapter 5/NestedTables/ProductNestedTableFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using Oracle.DataAccess.Client;
using Oracle.DataAccess.Types;


[OracleCustomTypeMappingAttribute("EDZEHOO.PRODUCTNESTEDTABLE")]
public class ProductNestedTableFactory : IOracleArrayTypeFactory
{
    public Array CreateArray(int numElems)
    {
        return new String[numElems];
    }
    public Array CreateStatusArray(int numElems)
    {
        return new OracleUdtStatus[numElems];
    }
}
